x-pack/plugin/sql/qa/mixed-node/src/test/java/org/elasticsearch/xpack/sql/qa/mixed_node/SqlCompatIT.java

```diff
@@ -154,42 +168,112 @@ public static String sqlQueryEntityWithOptionalMode(String query, Version bwcVer
             json.field("binary_format", false);
             json.field("version", bwcVersion.toString());
         }
+        if (fetchSize != null) {
+            json.field("fetch_size", fetchSize);
+        }
         json.endObject();
 
         return Strings.toString(json);
     }
 
-    public void testCursorFromOldNodeFailsOnNewNode() throws IOException {
-        assertCursorNotCompatibleAcrossVersions(bwcVersion, oldNodesClient, Version.CURRENT, newNodesClient);
+    public void testCursorFromOldNodeWorksOnNewNode() throws IOException {
+        assertCursorCompatibleAcrossVersions(bwcVersion, oldNodesClient, newNodesClient);
     }
 
-    @AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/83726")
-    public void testCursorFromNewNodeFailsOnOldNode() throws IOException {
-        assertCursorNotCompatibleAcrossVersions(Version.CURRENT, newNodesClient, bwcVersion, oldNodesClient);
+    public void testCursorFromNewNodeWorksOnOldNode() throws IOException {
+        assertCursorCompatibleAcrossVersions(Version.CURRENT, newNodesClient, oldNodesClient);
     }
 
-    private void assertCursorNotCompatibleAcrossVersions(Version version1, RestClient client1, Version version2, RestClient client2)
-        throws IOException {
+    private void assertCursorCompatibleAcrossVersions(Version version1, RestClient client1, RestClient client2) throws IOException {
         indexDocs();
 
+        String cursor = fetch1RecordAndReturnCursor(version1, client1);
+        assertCursorCanFetchDocs(client2, cursor);
+    }
+
+    public void testCursorFromOldNodeCanCloseOnNewNode() throws IOException {
+        assertCursorCloseWorksAcrossVersions(bwcVersion, oldNodesClient, newNodesClient);
+    }
+
+    public void testCursorFromNewNodeCanCloseOnOldNode() throws IOException {
+        assertCursorCloseWorksAcrossVersions(Version.CURRENT, newNodesClient, oldNodesClient);
+    }
+
+    private void assertCursorCloseWorksAcrossVersions(Version version1, RestClient client1, RestClient client2) throws IOException {
+        indexDocs();
+
+        String cursor = fetch1RecordAndReturnCursor(version1, client1);
+
+        Request scrollReq = new Request("POST", "_sql/close");
+        scrollReq.addParameter("error_trace", "true");
+        scrollReq.setJsonEntity("{\"cursor\": \"%s\"}".formatted(cursor));
+        Map<String, Object> scrollJson = performRequestAndReadBodyAsJson(client2, scrollReq);
+
+        assertEquals(scrollJson.get("succeeded"), true);
+    }
+
+    public void testCursorForIndexAllocatedOnNewNodeWorkOnOldNode() throws IOException {
```

Comment:
Unfortunately, this test and `testTextCursorFromNewNodeWorksOnOldNode` can only be tested once we have a release including the cursor redirect. Any ideas on how to run the test anyway would be highly appreciated :)

Reply:
Since 8.2 has now been branched out, I can backport this PR to 8.2 when it's ready to merge, see how the tests are doing and then merge to master.